src/coreclr/src/jit/morph.cpp

```diff
@@ -6345,7 +6345,9 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
 
                 var_types handleTyp = (var_types)(varTypeIsGC(tree->TypeGet()) ? TYP_BYREF : TYP_I_IMPL);
                 GenTree*  op1       = gtNewOperNode(GT_IND, handleTyp, addr);
-                op1->gtFlags |= GTF_IND_INVARIANT;
+                op1->gtFlags |= GTF_IND_NONFAULTING;
+
+                assert(!"Unreached GTF_IND_INVARIANT");
```

Comment:
Ditto for this, if it's now unreachable, why not remove it all?

Reply:
I will add an assert and remove the unreachable block,